File: categorize_new_members.js
function openOrSenior(data){
  var result = [];
  for (var i=0; i<data.length; i++) {
    if (data[i][0] < 55 || data[i][1] <= 7) {
      result.push('Open');
    } else {
      result.push('Senior');
    }
  }
  return result;
}

// function openOrSenior(data){
//   return data.map(([age, handicap]) => (age > 54 && handicap > 7) ? 'Senior' : 'Open');
// }

File: categorize_new_members.rb
def openOrSenior(data)
  data.map { |age, hc| ( age < 55 || hc <= 7 ? 'Open' : 'Senior' ) }
end

def openOrSenior(data)
    result = []
    data.each do |age, hc|
        if age < 55 || hc <= 7
            result << 'Open'
        else
            result << 'Senior'
        end
    end
    return result
end